osidb/models.py

```diff
@@ -1538,22 +1541,28 @@ def __init__(self, *args, **kwargs):
         Flaw, on_delete=models.CASCADE, related_name="snippets", blank=True, null=True
     )
 
-    def convert_snippet_to_flaw(self) -> Union[Flaw, None]:
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                name="unique_snippets", fields=["source", "external_id"]
+            ),
+        ]
+
+    def convert_nvd_snippet_to_flaw(self) -> Union[Flaw, None]:
         """
         Creates a new flaw from the snippet's content if a flaw with the given cve_id
         does not exist, and links them together. If a flaw already exists and does not
         contain the snippet of the current source, the snippet will be linked to it.
 
         Returns a flaw if it was newly created, None otherwise.
         """
-        cve_id = self.content["cve_id"]
+        cve_id = self.external_id
         created_flaw = None
 
-        if not cve_id or not Flaw.objects.filter(cve_id=cve_id):
+        flaw = Flaw.objects.filter(cve_id=cve_id).first()
+        if not flaw:
```

Comment:
The updated condition does not seem correct to me. In the case there is no CVE ID in the Snippet content we should not just blindly pair it with the first CVE-less flaw we have in the DB.